### ISSUE TITLE
plotting.plot_matrix requires labels argument to be provided

### ISSUE THREAD

**Issue by bthirion** [nilearn/nilearn#2049]
The following example breaks
```
from nilearn import datasets
atlas = datasets.fetch_atlas_msdl()
atlas_filename = atlas['maps']
labels = atlas['labels']
data = datasets.fetch_adhd(n_subjects=1)
from nilearn.input_data import NiftiMapsMasker
masker = NiftiMapsMasker(maps_img=atlas_filename, standardize=True,
                         memory='nilearn_cache', verbose=5)
time_series = masker.fit_transform(data.func[0],
                                   confounds=data.confounds)
from nilearn.connectome import ConnectivityMeasure
correlation_measure = ConnectivityMeasure(kind='correlation')
correlation_matrix = correlation_measure.fit_transform([time_series])[0]
import numpy as np
from nilearn import plotting
np.fill_diagonal(correlation_matrix, 0)
plotting.plot_matrix(correlation_matrix, labels=[], colorbar=True,
                     vmax=0.8, vmin=-0.8)
```

By contrast replacing the last line by
```
plotting.plot_matrix(correlation_matrix, labels=labels, colorbar=True,
                     vmax=0.8, vmin=-0.8)
```

The documentation of plot_matrix should be revised (labels is currently optional) or this should be fixed.

**Comment by kchawla-pi** [comment]
Is it related to the MatplotlibDeprecationWarning of always using unique axes or axes labels?

```
MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently
reuses the earlier instance. In a future version, a new instance will always be created and returned.
Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to
each axes instance. warnings.warn(message, mplDeprecation, stacklevel=1)  
```

**Comment by bthirion** [comment]
AFAIK this is not the same thing.
Note that setting `auto_fit=False` solves the issue.

**Comment by mjboos** [comment]
Hi,
the problem was in the logic of the if statements, since an empty list does evaluate to False, but not if directly queried (i.e. `not []` is unequal to `[] is False`).
I will go ahead and do a pull request.

**Comment by jeromedockes** [comment]
fixed in #2083
